lang/java/avro/src/main/java/org/apache/avro/generic/GenericData.java

```diff
@@ -178,6 +179,29 @@ public void addLogicalTypeConversion(Conversion<?> conversion) {
     return (Conversion<Object>) conversions.get(logicalType.getName());
   }
 
+  public static final String FAST_READER_PROP = "org.apache.avro.fastread";
+  private boolean fastReaderEnabled = "true".equalsIgnoreCase(System.getProperty(FAST_READER_PROP));
+  private FastReaderBuilder fastReaderBuilder = null;
+
+  public GenericData setFastReaderEnabled(boolean flag) {
+    this.fastReaderEnabled = flag;
+    return this;
+  }
+
+  public boolean isFastReaderEnabled() {
+    return fastReaderEnabled && FastReaderBuilder.isSupportedData(this);
+  }
+
+  public FastReaderBuilder getFastReaderBuilder() {
+    if (fastReaderBuilder == null) {
+      FastReaderBuilder localFastReader = new FastReaderBuilder(this);
```

Comment:
Definitely a nit, but why introduce the local variable here?

Reply:
I can sure shorten this. Yes, not quite sure why I did it in the first place.

Reply:
Fixed.